firebase/test/core/data/schemas/local-public-collection.spec.ts

```diff
@@ -0,0 +1,87 @@
+import Ajv2020 from "ajv/dist/2020";
+import { SchemaValidator } from "#data/schemas/schema-validator";
+import { doesNotThrow, throws } from "assert";
+import SerializationError from "#faults/errors/serialization-error";
+import { newRawLocalCollection, newRawMemo } from "./collections-fakes";
+
+describe("Local Public Collection Schema Validation", () => {
+  const validator = new SchemaValidator(new Ajv2020());
+  const requiredProperties = ["memos"];
+  const optionalProperties: string[] = [];
+  const arrayProperties = ["memos"];
+  const stringProperties: string[] = [];
+  // Maps an array property to a list of repeated items.
+  const nonRepeatableArrayProperties = new Map<string, any[]>([["memos", [newRawMemo(), newRawMemo()]]]);
+  // Maps a property to a type that couldn't be associated to it.
+  const incorrectPropertiesTypes = new Map<string, any>([["memos", "string"]]);
+
+  it("should validate raw collection structure", () => {
+    const rawCollection = newRawLocalCollection();
+
+    doesNotThrow(() => validator.validateObject("local-public-collection", rawCollection));
+  });
+
+  it("should throw when a required property is not set", () => {
+    for (const requiredProperty of requiredProperties) {
+      const rawCollection = newRawLocalCollection();
+
+      delete rawCollection[requiredProperty];
+
+      throws(() => validator.validateObject("local-public-collection", rawCollection), SerializationError);
+    }
+  });
+
+  it("should not throw when a optional property is not set", () => {
+    for (const optionalProperty of optionalProperties) {
+      const rawCollection = newRawLocalCollection();
+
+      delete rawCollection[optionalProperty];
+
+      doesNotThrow(() => validator.validateObject("local-public-collection", rawCollection), SerializationError);
```

Comment:
does not throw with a `SerializationError`?

Reply:
Removed